### DISCUSSION TITLE
How do I define global variables and reference them from markdown?

### DISCUSSION THREAD

**Discussion by jrtitus** [vuepress/core#1164] [Q&A] [answered]
For example, I have the beginning of a URL that contains a version number that I would like to be able to update in a single location and use it such as `[link text]({{ urlPrefixWithVersion }}/rest/of/the/url.html)`. Is it possible?

**Comment by Mister-Hope** [comment]


**Reply by jrtitus** [reply]
Where would I define the variable in the configuration globally? I tried adding it to the `define` object in the user config, but I get `undefined` when trying to reference it using the `<a>` tag. Am I missing something obvious in the documentation? 😅 

The component definition works well. Thank you for that.

**Reply by Mister-Hope** [reply]
Maybe a bug I guess, we have some issues with markdown define variables not far ago. Could you try to create a reproduction with our https://v2.vuepress.vuejs.org/new ?

**Reply by jrtitus** [reply]
Sorry for the delay. Here's an example showing the `undefined` variable in markdown using the `<a>` tag: https://stackblitz.com/edit/vuepress-nxr9gg?file=docs/guide/README.md

**Comment by Mister-Hope** [comment, marked as answer]
Here is a full demo:

https://stackblitz.com/edit/vuepress-markdown-variable?file=docs%2F.vuepress%2Fconfig.ts,docs%2F.vuepress%2Fclient.ts,docs%2Fguide%2FREADME.md

**Reply by jrtitus** [reply]
Hmm.. OK. I wasn't expecting it to be necessary to add the script tag to reference the defines.

Your example would make for a good addition to the documentation for the usage of `define`. Thank you for the assistance!
